### PR TITLE
ci/permissions: Restrict permissions for remaining workflows

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @mmorel-35 - not sure if we need to add some back in jobs

**Comment by mmorel-35** [comment]
What do you mean by "back in jobs" ?

**Comment by phlax** [comment]
> What do you mean by "back in jobs" ?

provide job-level permissions

**Comment by mmorel-35** [comment]
OK, Usually, I'm using https://app.stepsecurity.io/secureworkflow to help defining the right configuration : 

image

Is this also what you did ?

**Comment by phlax** [comment]
> Is this also what you did ?

no - i generally define them by hand (after checking the docs and the action)

in this case im responding to a scorecard alert that these wfs dont have global token restrictions

File: .github/workflows/envoy-sync.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 name: 'Sync downstream'
 
+permissions:
+  contents: read
```
Comment (phlax) [inline .github/workflows/envoy-sync.yml:4]: this one is good - it uses appauth

**Comment by phlax** [comment]
afaict - this is what is required for codeql

```yaml
    permissions:
      contents: read
      security-events: write
      pull-requests: read
```

https://github.com/github/codeql-action/issues/464

**Comment by phlax** [comment]
which i think is already covered - except perhaps read perms in a private repo - ill update

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**